keps/sig-storage/3476-volume-group-snapshot/README.md

```diff
@@ -0,0 +1,1660 @@
+        // for snapshotting.
+        // This labelSelector will be used to match the label added to a PVC.
+        Selector *metav1.LabelSelector
+
+        // VolumeGroupSnapshotSecretRef is a reference to the secret object containing
+        // sensitive information to pass to the CSI driver to complete the CSI
+        // calls for VolumeGroupSnapshots.
+        // This field is optional, and may be empty if no secret is required. If the
+        // secret object contains more than one secret, all secrets are passed.
+        // +optional
+        VolumeGroupSnapshotSecretRef *SecretReference
+}
+
+Type VolumeGroupSnapshotStatus struct {
+        // +optional
+        BoundVolumeGroupSnapshotContentName *string
+
+        // ReadyToUse becomes true when ReadyToUse on all individual snapshots become true
+        // +optional
+        ReadyToUse *bool
+
+        // +optional
+        CreationTime *metav1.Time
+
+        // +optional
+        Error *VolumeGroupSnapshotError
+
+        // List of volume snapshots
+        // +optional
+        SnapshotList []VolumeSnapshot
+}
+
+// Describes an error encountered on the group snapshot
+type VolumeGroupSnapshotError struct {
+        // time is the timestamp when the error was encountered.
+        // +optional
+        Time *metav1.Time
+
+        // message details the encountered error
+        // +optional
+        Message *string
+}
+```
+
+#### VolumeGroupSnapshotContent
+
+```
+// VolumeGroupSnapshotContent
+type VolumeGroupSnapshotContent struct {
+        metav1.TypeMeta `json:",inline"`
+        // Standard object's metadata.
+        // +optional
+        metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+        // Spec defines the desired characteristics of a group snapshot content
+        Spec VolumeGroupSnapshotContentSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+
+        // Status represents the latest observed state of the group snapshot content
+        // +optional
+        Status *VolumeGroupSnapshotContentStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// VolumeGroupSnapshotContentSpec describes the common attributes of a group snapshot content
+type VolumeGroupSnapshotContentSpec struct {
+        // Required
+        // VolumeGroupSnapshotRef specifies the VolumeGroupSnapshot object
+        // to which this VolumeGroupSnapshotContent object is bound.
+        VolumeGroupSnapshotRef core_v1.ObjectReference
+
+        // Required
+        VolumeGroupSnapshotDeletionPolicy VolumeGroupSnapshotDeletionPolicy
+
+        // Required
+        Driver string
+
+        // +optional
+        VolumeGroupSnapshotClassName *string
+
+        // Required
+        Source VolumeGroupSnapshotContentSource
+}
+
+// OneOf
+type VolumeGroupSnapshotContentSource struct {
+        // Dynamical provisioning of VolumeGroupSnapshot
+        // +optional
+        VolumeGroupHandle *string
```

Comment:
Is this still needed?

Reply:
Oh, I forgot to change this.  I changed to the following.  We still need to have PV names to find out the source_volume_ids to pass to the CreateVolumeGroupSnapshot CSI call.

```
        // Dynamical provisioning of VolumeGroupSnapshot
        // A list of PersistentVolume names to be snapshotted together
        // +optional
        PersistentVolumeNames []string
```

Reply:
Done